testing/src/test/resources/io/split/client/testing/cucumber/Split.feature

```diff
@@ -0,0 +1,19 @@
+@split[cappuccino:off]
+@split[dollars:on]
+Feature: Split
```

Comment:
Would you mind educating me how the unit test binds to the split listed in this file?

also, is the idea to have a single descritptor file or one per test class or test case?

Reply:
I'm not sure what you mean by unit test or descriptor file. I'll clarify some of the terminology used by Cucumber (which is closer to an acceptance testing tool than a unit testing tool).

A `.feature` file (sometimes called *gherkin document*) contains multiple *scenarios* (sometimes called *examples*).
Each *Scenario* is an acceptance test and consists of the following parts:

* One or more [Steps](https://cucumber.io/docs/gherkin/reference/#example) (`Given`, `When`, `Then`).
* [Step Definitions](https://cucumber.io/docs/cucumber/step-definitions/), which are Java methods annotated with `@Given`, `When` or `@Then`.

Cucumber creates a new instance of every class with step definitions before the scenario runs. This is similar to how JUnit creates a new instance of a test class for every `@Test` method it executes.

For each step in the scenario, Cucumber then finds a matching step definition and invokes the method.

Step definition classes aren't scoped to a particular `.feature` file. The steps in a `.feature` file can invoke step definitions from any step definition class - the matching is based on the expression in the step definition annotation.

Scenarios can have [tags](https://cucumber.io/docs/cucumber/api/?sbsearch=tag#tags). The code in this library looks for tags of the format `@split[feature:treatment]` and uses that to register the appropriate treatments.